apps/mymedicare_cb/views.py

```diff
@@ -234,14 +269,36 @@ def generate_nonce(length=26):
 
 @never_cache
 def mymedicare_login(request):
-    redirect = settings.MEDICARE_REDIRECT_URI
-    mymedicare_login_url = settings.MEDICARE_LOGIN_URI
+    # SLS vs. SLSx flow based on feature switch slsx-enable (true = SLSx / false = SLS).
+    if waffle.switch_is_active('slsx-enable'):
+        redirect = settings.MEDICARE_SLSX_REDIRECT_URI
+        mymedicare_login_url = settings.MEDICARE_SLSX_LOGIN_URI
+
+        # Get auth flow session values.
+        auth_flow_dict = get_session_auth_flow_trace(request)
+
+        # Perform health check on SLSx service
+        slsx_client = OAuth2ConfigSLSx()
+        try:
+            slsx_client.service_health_check()
+        except requests.exceptions.HTTPError as e:
+            log_authenticate_start(auth_flow_dict, "FAIL",
+                                   "SLSx service health check error {reason}".format(reason=e))
+            raise BBSLSxHealthCheckFailedException(settings.MEDICARE_ERROR_MSG)
```

Comment:
If the SLSx health check end point has a non 2xx result code response, this prevents the beneficiary from being directed to the Medicare.gov (via SLSx /sso/authorize/) site to login with username and password.

Is this a good way to handle this?  Or should the redirect occur anyway and just use this health check information for logging purposes?  Are there any potential issues doing this?

For example, can we always assume that the login is also having issues, if the health check is failing?

Reply:
I think this is probably safe given if the healthcheck is down, something else with SLSx isn't working as expected.

Reply:
I do have the same comment here, about whether this exception check could be consolidated into the SLSx config client class.